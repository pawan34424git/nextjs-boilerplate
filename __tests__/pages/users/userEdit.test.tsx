import UserEditPage, { getServerSideProps } from "@/pages/users/[userId]";
import "@testing-library/jest-dom";
import { GetServerSidePropsContext } from "next";
import renderer from "react-test-renderer";

describe("User Edit", () => {
  it("user edit rendering", () => {
    const props = {
      GOOGLE_ANALYTICS_ID: "test",
      query: { userId: "test" },
      profile: "testing",
    };

    const tree = renderer.create(<UserEditPage {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("check serverSideProps", async () => {
    const req: any = {};
    const res: any = {};
    const nextParams: GetServerSidePropsContext = {
      req,
      res,
      resolvedUrl: "",
      query: {
        userId: "test",
      },
    };
    const { props } = await getServerSideProps(nextParams);
    expect(props.query.userId).toBe("test");
  });
});
