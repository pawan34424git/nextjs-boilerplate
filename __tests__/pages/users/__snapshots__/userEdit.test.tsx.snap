// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Edit user edit rendering 1`] = `
<div
  className="min-h-full"
>
  <main>
    <div
      className="flex-col rounded m-0 p-0 dark:bg-slate-800  shadow-inner px-2 py-1 "
    >
      <h1>
        User Id: 
        test
      </h1>
      <h1>
        GOOGLE ID: 
        test
      </h1>
      <h1>
        Profile: 
        testing
      </h1>
    </div>
  </main>
</div>
`;
